### PR TITLE
Generate Type class code.

### PR DESCRIPTION
**Pull request by dlibenzi**

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -48,6 +48,12 @@ def _check_env_flag(name, default=''):
 
 base_dir = os.path.dirname(os.path.abspath(__file__))
 
+generate_code_cmd = [os.path.join(base_dir, 'scripts', 'generate_code.sh')]
```
Comment (asuhan) [inline setup.py]: You need to `pip install lark-parser` in `.circleci/build.sh`.

File: scripts/generate_code.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+CDIR=$(dirname $0)
+XDIR=$(realpath $CDIR/..)
+PTDIR=$(realpath $XDIR/..)
+
+$CDIR/gen.py \
```
Comment (asuhan) [inline scripts/generate_code.sh]: We have to run it in the Python where we install lark.

File: scripts/generate_code.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+CDIR=$(dirname $0)
+XDIR=$(realpath $CDIR/..)
```
Comment (asuhan) [inline scripts/generate_code.sh]: `realpath` is not available in the CircleCI environment.

File: .circleci/build.sh [hunk 4]
```diff
@@ -30,6 +30,9 @@ cd $PYTORCH_DIR
 # Install ninja to speedup the build
 pip install ninja
 
+# Install the Lark parser required for the XLA->ATEN Type code generation.
+pip install lark-parser
```
[the remark this answers is not in the document]
Comment (dlibenzi) [reply .circleci/build.sh]: Yup.

File: scripts/generate_code.sh (hunk 3 above)
Comment (dlibenzi) [reply scripts/generate_code.sh]: Gone.

File: scripts/generate_code.sh (hunk 2 above)
Comment (dlibenzi) [reply scripts/generate_code.sh]: Done.

**Comment by asuhan** [comment]
Tests should pass once you rebase, it's missing https://github.com/pytorch/xla/commit/382df7f866191d4d036c23a43024d023da25c6ed.

**Comment by dlibenzi** [comment]
Whoops ... I missed Daniel's hack :|
Let me know if you see an issue Daniel.

**Comment by dlibenzi** [comment]
Sorry about that! :|
Will remember next time!

**Pull request merged**